### PR TITLE
Updating Compose documentation

### PR DESCRIPTION
**Pull request by moxiegirl**
- Split out commands into individual pages for maintainability
- Add full usage in commands pages for usability
- Updated description of the run command
- Carries and expands on doc change in #1701

Signed-off-by: Mary Anthony <mary@docker.com>

ping @aanand

### CODE REVIEW

File: docs/reference/docker-compose.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+ Specify what file to read configuration from. If not provided, Compose will look
+ for `docker-compose.yml` in the current working directory, and then each parent
+ directory successively, until found.
+
+ Use a `-` as the filename to read configuration from stdin. When stdin is used
+ all paths in the configuration will be relative to the current working
+ directory.
+
+ Specifies an alternate project name (default: current directory name)
```
Comment (aanand) [inline docs/reference/docker-compose.md]: Looks like the descriptions for the global `-f` and `-p` options have ended up here without any context.

File: docs/reference/run.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+--entrypoint CMD      Override the entrypoint of the image.
+-e KEY=VAL            Set an environment variable (can be used multiple times)
+-u, --user=""         Run as specified username or uid
+--no-deps             Don't start linked services.
+--rm                  Remove container after run. Ignored in detached mode.
+--service-ports       Run command with the service's ports enabled and mapped to the host.
+-T                    Disable pseudo-tty allocation. By default `docker-compose run` allocates a TTY.
+```
+
+Runs a one-time command against a service. For example, the following command starts the `web` service and runs `manage.py shell` in `python`.
```
Comment (aanand) [inline docs/reference/run.md]: "runs `manage.py shell` in `python`" feels a bit strange to me - makes it sound like `python` might be a container, or an image. Maybe:

> For example, the following command starts a container with the `web` service's image and configuration, and `python manage.py shell` as its command.

Also, I think `bash` would be a better example here, so people who don't know Django aren't thrown off wondering what `manage.py shell` is.

> For example, the following command starts a container with the `web` service's image and configuration, and `bash` as its command.
>
>     $ docker-compose run web bash

File: docs/reference/index.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+keywords = ["fig, composition, compose, docker, orchestration, cli,  reference"]
+[menu.main]
+identifier = "smn_compose_cli"
+parent = "smn_compose_ref"	
++++
+<![end-metadata]-->
+
+## Compose CLI reference
+
+The following pages describe the usage information for the Docker Compose subcommands. You can also see this information by running `docker-compose [SUBCOMMAND] --help` from the command line.
```
Comment (aanand) [inline docs/reference/index.md]: When we say "The following pages", where are these pages going to be listed? Is it just in the sidebar?

File: docs/reference/docker-compose.md (hunk 1 above)
Comment (moxiegirl) [inline docs/reference/docker-compose.md]: You are right. I'll fix that.

File: docs/reference/index.md (hunk 3 above)
Comment (moxiegirl) [inline docs/reference/index.md]: This is something sloppy on my part, I forgot to do this.

**Comment by moxiegirl** [comment]
comments incorporated. ping @aanand

**Comment by aanand** [comment]
LGTM

**Pull request merged**